src/scripts/install_machine.py

```diff
@@ -0,0 +1,323 @@
+        for value in graphics.find('enum').findall('value'):
+            consoles.append(value.text)
+
+    logging.debug('get_graphics_capabilies: ' + ', '.join(consoles))
+
+    return list(filter(lambda console: console in ['vnc', 'spice'], consoles))
+
+
+@contextmanager
+def prepare_unattended(args):
+    params = []
+    if args.get('unattended'):
+        params.append("--unattended")
+
+        unattended_params = []
+        unattended_params.append(f"profile={args['profile']}")
+
+        if 'rootPassword' in args:
+            root_pass_file = tempfile.NamedTemporaryFile(
+                prefix="cockpit-machines-",
+                suffix=("-admin-password"),
+                mode='w+'
+            )
+            root_pass_file.write(args['rootPassword'])
+            root_pass_file.flush()
+            unattended_params.append(f"admin-password-file={root_pass_file.name}")
+
+        if 'userLogin' in args:
+            unattended_params.append(f"user-login={args['userLogin']}")
+
+        if 'userPassword' in args:
+            user_pass_file = tempfile.NamedTemporaryFile(
+                prefix="cockpit-machines-",
+                suffix=("-user-password"),
+                mode='w+'
+            )
+            user_pass_file.write(args['userPassword'])
+            user_pass_file.flush()
+            unattended_params.append(f"user-password-file={user_pass_file.name}")
+
+        params.append(",".join(unattended_params))
+
+    yield params
+
+
+@contextmanager
+def prepare_cloud_init(args):
+    params = []
+    if args['sourceType'] == 'cloud':
+        params.append("--cloud-init")
+
+        content = "#cloud-config\n"
+        if 'userPassword' in args and 'userLogin' in args:
+            content += "users:\n"
+            content += f"  - name: {args['userLogin']}:{args['userPassword']}\n"
+
+        if 'rootPassword' in args:
+            content += "chpasswd:\n"
+            content += "  list: |\n"
+            content += f"    root:{args['rootPassword']}\n"
+            content += "  expire: False\n"
+
+            user_data_file = tempfile.NamedTemporaryFile(
+                prefix="cockpit-machines-",
+                suffix=("-user-data"),
+                mode='w+'
+            )
+            user_data_file.write(content)
+            user_data_file.flush()
+            params.append(f"user-data={user_data_file.name}")
+
+    yield params
+
+
+def prepare_installation_source(params, args):
+    only_define = False
+    if args['type'] == 'create':
+        if not args['startVm']:
+            only_define = True
+            params.append("--print-xml")
+
+    if args['sourceType'] == "pxe":
+        params += ['--pxe', '--network', args['source']]
+    elif args['sourceType'] == "os":
+        params += ['--install', f"os={args['os']}"]
+    elif args['sourceType'] in ['disk_image', 'cloud']:
+        params.append("--import")
+    elif ((args['source'][0] == '/' and os.path.isfile(args['source'])) or
```

Comment:
To be honest I just tried to translate the condition from the bash script here,  https://github.com/cockpit-project/cockpit-machines/blob/35c24110395eea8217c85f34eb5bc32d47dd536b/src/scripts/create_machine.sh#L132

This script can be definitely improved, but I tried to avoid actual functional changes and really stick to the rewrite to python, without affecting the functionality.